### PR TITLE
rename master to cluster_manager commit_2

### PR DESCRIPTION
**Pull request by kaushalmahi12**
Signed-off-by: Kaushal Kumar <kshkmr@amazon.com>

**Is your feature request related to a problem? Please provide an existing Issue # , or describe.**
[Master Nomenclature Issue](https://github.com/opensearch-project/performance-analyzer-rca/issues/139)

**Describe the solution you are proposing**

**Describe alternatives you've considered**

**Additional context**

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/performance-analyzer-rca/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

This PR is one of few That I will create against the issue mentioned in the description. There is parent [PR](https://github.com/opensearch-project/performance-analyzer-rca/pull/205) which is open but too long to review at once.

### CODE REVIEW

File: src/main/java/org/opensearch/performanceanalyzer/decisionmaker/deciders/Decision.java [hunk 1]
```diff
@@ -57,7 +57,8 @@ public boolean isEmpty() {
 
     @Override
     public FlowUnitMessage buildFlowUnitMessage(String graphNode, InstanceDetails.Id node) {
-        // All deciders run on the master node, (in initial versions), so we dont expect Decisions
+        // All deciders run on the cluster_manager node, (in initial versions), so we don't expect
+        // Decisions
```
Comment (kiranprakash154) [inline src/main/java/org/opensearch/performanceanalyzer/decisionmaker/deciders/Decision.java:61]: nit : can be a single line
```suggestion
        // All deciders run on the cluster_manager node, (in initial versions), so we don't expect Decisions
```

File: src/main/java/org/opensearch/performanceanalyzer/metrics/AllMetrics.java [hunk 2]
```diff
@@ -789,16 +789,17 @@ public String toString() {
         }
 
         public static class Constants {
-            public static final String PENDING_TASKS_COUNT_VALUE = "Master_PendingQueueSize";
+            public static final String PENDING_TASKS_COUNT_VALUE =
+                    "ClusterManager_PendingQueueSize";
```
Comment (kiranprakash154) [inline src/main/java/org/opensearch/performanceanalyzer/metrics/AllMetrics.java:793]: nit : can be a single line
```suggestion
            public static final String PENDING_TASKS_COUNT_VALUE = "ClusterManager_PendingQueueSize";
```

File: src/main/java/org/opensearch/performanceanalyzer/metrics/PerformanceAnalyzerMetrics.java [hunk 3]
```diff
@@ -33,17 +33,18 @@ public class PerformanceAnalyzerMetrics {
...
+    public static final String sClusterManagerClusterUpdate = "cluster_manager_cluster_update";
     public static final String sElectionTermPath = "election_term";
     public static final String sHttpPath = "http";
     public static final String sOSPath = "os_metrics";
     public static final String sHeapPath = "heap_metrics";
     public static final String sNodesPath = "node_metrics";
     public static final String sPendingTasksPath = "pending_tasks";
-    public static final String sMasterThrottledTasksPath = "master_throttling_metrics";
+    public static final String sClusterManagerThrottledTasksPath =
+            "cluster_manager_throttling_metrics";
```
Comment (kiranprakash154) [inline src/main/java/org/opensearch/performanceanalyzer/metrics/PerformanceAnalyzerMetrics.java:47]: nit : can be a single line
```suggestion
    public static final String sClusterManagerThrottledTasksPath = "cluster_manager_throttling_metrics";
```

File: src/main/java/org/opensearch/performanceanalyzer/reader/ClusterDetailsEventProcessor.java [hunk 4]
```diff
@@ -226,18 +229,20 @@ public String getRole() {
...
-                        RcaControllerHelper.getElectedMasterHostAddress();
-                isMasterNode = this.hostAddress.equalsIgnoreCase(electedMasterHostAddress);
+            // query cat cluster_manager api directly to read the node's role in case the
+            // ClusterDetailsEvent
+            // received from writer does not have "IS_CLUSTER_MANAGER_NODE" field.
+            if (isClusterManager == null) {
+                final String electedClusterManagerHostAddress =
+                        RcaControllerHelper.getElectedClusterManagerHostAddress();
+                isClusterManager =
+                        this.hostAddress.equalsIgnoreCase(electedClusterManagerHostAddress);
```
Comment (kiranprakash154) [inline src/main/java/org/opensearch/performanceanalyzer/reader/ClusterDetailsEventProcessor.java:243]: nit : can be a single line
```suggestion
                isClusterManager = this.hostAddress.equalsIgnoreCase(electedClusterManagerHostAddress);
```

File: src/main/java/org/opensearch/performanceanalyzer/reader/ClusterManagerEventMetricsSnapshot.java [hunk 5]
```diff
@@ -25,8 +25,9 @@
 import org.opensearch.performanceanalyzer.metrics.MetricsConfiguration;
 import org.opensearch.performanceanalyzer.metricsdb.MetricsDB;
 
-public class MasterEventMetricsSnapshot implements Removable {
-    private static final Logger LOG = LogManager.getLogger(MasterEventMetricsSnapshot.class);
+public class ClusterManagerEventMetricsSnapshot implements Removable {
+    private static final Logger LOG =
+            LogManager.getLogger(ClusterManagerEventMetricsSnapshot.class);
```
Comment (kiranprakash154) [inline src/main/java/org/opensearch/performanceanalyzer/reader/ClusterManagerEventMetricsSnapshot.java:30]: nit : can be a single line

```suggestion
    private static final Logger LOG = LogManager.getLogger(ClusterManagerEventMetricsSnapshot.class);
```

File: src/main/java/org/opensearch/performanceanalyzer/reader/ClusterManagerEventMetricsSnapshot.java [hunk 6]
```diff
@@ -66,35 +67,36 @@ public MasterEventMetricsSnapshot(Connection conn, Long windowStartTime) {
...
+                                                AllMetrics.ClusterManagerMetricDimensions
+                                                        .CLUSTER_MANAGER_TASK_PRIORITY
                                                         .toString()),
                                         String.class));
                         this.add(
                                 DSL.field(
                                         DSL.name(
-                                                AllMetrics.MasterMetricDimensions.MASTER_TASK_TYPE
+                                                AllMetrics.ClusterManagerMetricDimensions
+                                                        .CLUSTER_MANAGER_TASK_TYPE
```
Comment (kiranprakash154) [inline src/main/java/org/opensearch/performanceanalyzer/reader/ClusterManagerEventMetricsSnapshot.java:85]: ❤️

File: src/main/java/org/opensearch/performanceanalyzer/reader/ClusterManagerEventMetricsSnapshot.java [hunk 7]
```diff
@@ -217,142 +220,151 @@ public Result<Record> fetchQueueAndRunTime() {
                         this.add(
                                 DSL.field(
                                         DSL.name(
-                                                AllMetrics.MasterMetricDimensions
-                                                        .MASTER_TASK_INSERT_ORDER
+                                                AllMetrics.ClusterManagerMetricDimensions
+                                                        .CLUSTER_MANAGER_TASK_INSERT_ORDER
```
Comment (kiranprakash154) [inline src/main/java/org/opensearch/performanceanalyzer/reader/ClusterManagerEventMetricsSnapshot.java:224]: Lets maintain the same pattern through out this file.

```java
AllMetrics
.ClusterManagerMetricDimensions
.CLUSTER_MANAGER_TASK_INSERT_ORDER
```

or 

```java
AllMetrics.ClusterManagerMetricDimensions
.CLUSTER_MANAGER_TASK_INSERT_ORDER
```

File: src/main/java/org/opensearch/performanceanalyzer/reader/MetricPropertiesConfig.java [hunk 8]
```diff
@@ -305,12 +306,13 @@ private MetricPropertiesConfig() {
...
-                MetricName.MASTER_CLUSTER_UPDATE_STATS,
+                MetricName.CLUSTER_MANAGER_CLUSTER_UPDATE_STATS,
                 new MetricProperties(
                         MetricProperties.EMPTY_DIMENSION,
-                        AllMetrics.MasterClusterUpdateStatsValue.values(),
+                        AllMetrics.ClusterManagerClusterUpdateStatsValue.values(),
                         createFileHandler(
-                                metricPathMap.get(MetricName.MASTER_CLUSTER_UPDATE_STATS))));
+                                metricPathMap.get(
+                                        MetricName.CLUSTER_MANAGER_CLUSTER_UPDATE_STATS))));
```
Comment (kiranprakash154) [inline src/main/java/org/opensearch/performanceanalyzer/reader/MetricPropertiesConfig.java:315]: nit : can be a single line
```suggestion
                                metricPathMap.get(MetricName.CLUSTER_MANAGER_CLUSTER_UPDATE_STATS))));
```

File: src/main/java/org/opensearch/performanceanalyzer/reader/ReaderMetricsProcessor.java [hunk 9]
```diff
@@ -60,11 +60,12 @@ public class ReaderMetricsProcessor implements Runnable {
...
     private NavigableMap<Long, ShardRequestMetricsSnapshot> shardRqMetricsMap;
     private NavigableMap<Long, HttpRequestMetricsSnapshot> httpRqMetricsMap;
-    private NavigableMap<Long, MasterEventMetricsSnapshot> masterEventMetricsMap;
+    private NavigableMap<Long, ClusterManagerEventMetricsSnapshot> clusterManagerEventMetricsMap;
     private NavigableMap<Long, GarbageCollectorInfoSnapshot> gcInfoMap;
     private Map<AllMetrics.MetricName, NavigableMap<Long, MemoryDBSnapshot>> nodeMetricsMap;
     private NavigableMap<Long, FaultDetectionMetricsSnapshot> faultDetectionMetricsMap;
-    private NavigableMap<Long, MasterThrottlingMetricsSnapshot> masterThrottlingMetricsMap;
+    private NavigableMap<Long, ClusterManagerThrottlingMetricsSnapshot>
+            clusterManagerThrottlingMetricsMap;
```
Comment (kiranprakash154) [inline src/main/java/org/opensearch/performanceanalyzer/reader/ReaderMetricsProcessor.java:68]: nit : can be a single line
```suggestion
    private NavigableMap<Long, ClusterManagerThrottlingMetricsSnapshot> clusterManagerThrottlingMetricsMap;
```

File: src/main/java/org/opensearch/performanceanalyzer/reader/ReaderMetricsProcessor.java [hunk 10]
```diff
@@ -516,26 +519,27 @@ private void emitShardRequestMetrics(
...
-        if (masterEventMetricsMap.containsKey(prevWindowStartTime)) {
+        if (clusterManagerEventMetricsMap.containsKey(prevWindowStartTime)) {
 
-            MasterEventMetricsSnapshot preMasterEventSnapshot =
-                    masterEventMetricsMap.get(prevWindowStartTime);
-            MetricsEmitter.emitMasterEventMetrics(metricsDB, preMasterEventSnapshot);
+            ClusterManagerEventMetricsSnapshot preClusterManagerEventSnapshot =
+                    clusterManagerEventMetricsMap.get(prevWindowStartTime);
+            MetricsEmitter.emitClusterManagerEventMetrics(
+                    metricsDB, preClusterManagerEventSnapshot);
```
Comment (kiranprakash154) [inline src/main/java/org/opensearch/performanceanalyzer/reader/ReaderMetricsProcessor.java:529]: nit : can be a single line
```suggestion
            MetricsEmitter.emitClusterManagerEventMetrics(metricsDB, preClusterManagerEventSnapshot);
```

**Review by kiranprakash154 — APPROVED** [review]
some nits, 
but i don't see any change in logic what so ever, so i think we should be good.

File: src/main/java/org/opensearch/performanceanalyzer/decisionmaker/deciders/Decision.java (hunk 1 above)
Comment (kaushalmahi12) [reply src/main/java/org/opensearch/performanceanalyzer/decisionmaker/deciders/Decision.java:61]: This change is amended in the last commit.

**Pull request closed without merging**